lightningd/peer_htlcs.c

```diff
@@ -621,11 +621,45 @@ static void forward_htlc(struct htlc_in *hin,
 {
 	const u8 *failmsg;
 	struct lightningd *ld = hin->key.channel->peer->ld;
-	struct channel *next = active_channel_by_scid(ld, scid);
+	struct channel *next;
 	struct htlc_out *hout = NULL;
 
+	/* This is a shortcut for specifying next peer; doesn't mean
+	 * the actual channel! */
+	next = any_channel_by_scid(ld, scid);
+	if (next) {
+		struct peer *peer = next->peer;
+		struct channel *channel;
+		struct amount_msat best_spendable = channel_amount_spendable(next);
+
+		/* Seek largest capacity channel! */
+		list_for_each(&peer->channels, channel, list) {
+			struct amount_msat spendable;
+			if (!channel_active(channel))
+				continue;
+			spendable = channel_amount_spendable(channel);
+			if (!amount_msat_greater(spendable, best_spendable))
+				continue;
+
+			/* Don't override if fees differ... */
+			if (channel->feerate_base != next->feerate_base
+			    || channel->feerate_ppm != next->feerate_ppm)
```

Comment:
Could relax the equality to allow forwarding if the channel we're picking would end up cheaper than the one indicated by the scid.

Reply:
Yes; generally though you can use a slight differentiator to remove this heuristic.  I'm not sure what the Right Thing is here, TBH.